### DISCUSSION TITLE
Seekable streams for attachments

### DISCUSSION THREAD

**Discussion by BickelLukas** [ravendb/ravendb#19366] [Q&A] [answered]
We are storing large attachments in the database (video files). These files need to be then streamed back to the user  from our AspNetCore App with `enableRangeProcessing` set to true in order for video playback to work correctly.

The ravendb attachment stream is not seekable and we have to copy the stream to a MemoryStream before serving. This causes problems for large video files. 

Is there any option to pass the stream directly from the database where rangeProcessing works as intended? 

Example: 
```cs
public class VideoController(IAttachmentUrlGenerator generator, IAsyncDocumentSession session) : ControllerBase
{
    public async Task<IActionResult> Stream(string documentId)
    {
        var attachment = await session.Advanced.Attachments.GetAsync(documentId, "video.mp4");

        // we don't want to load the complete attachment but rather pipe it straight through to the client
        var ms = new MemoryStream();
        await attachment.Stream.CopyToAsync(ms);
        return File(ms, attachment.Details.ContentType, attachment.Details.Name, null, new EntityTagHeaderValue($@"""{attachment.Details.Hash}"""), enableRangeProcessing: true);
    }
}
```

**Comment by ppekrol** [comment, marked as answer]
At the moment we do not support seekable streams for attachments. There is an open issue for that https://issues.hibernatingrhinos.com/issue/RavenDB-19049/Add-support-for-seekable-attachments but without any schedule at this point.

**Reply by BickelLukas** [reply]
@ppekrol I just looked at the code, the underlying VoronStream looks to be seekable. I'd be happy to give it a try to contribute this feature if you're open to it.

**Reply by ppekrol** [reply]
Of course, we would be glad to accept such a contribution.

**Reply by BickelLukas** [reply]
I implemented the feature and created a pull request: https://github.com/ravendb/ravendb/pull/19378
